kura/org.eclipse.kura.linux.position/src/main/java/org/eclipse/kura/linux/position/NMEAParser.java

```diff
@@ -207,46 +206,37 @@ public void parseSentence(String sentence) {
         }
     }
 
-    double convertPositionlat(String pos, String direction) {
-        double floatLatDegrees = 0;
-        double floatLatMinutes = 0;
-        String s;
-
+    /**
+     * @param pos
+     *            DDD?MM?.dddd
+     * @param direction
+     *            N/S, E/W
+     * @param degChars
+     *            number of characters representing degrees
+     * @return
+     */
+    private double convertPosition(String pos, String direction, int degChars) {
         if (pos.length() < 6) {
             return 0;
         }
 
-        // This copies the arrays to temporary arrays
-        s = pos.substring(0, 2);
-        floatLatDegrees = Double.parseDouble(s);
-        s = pos.substring(2);
-        floatLatMinutes = Double.parseDouble(s);
-        floatLatDegrees = floatLatDegrees + floatLatMinutes / 60;
-        if (direction.contains("S")) {
-            floatLatDegrees = floatLatDegrees * -1;
+        String s = pos.substring(0, degChars);
+        double deg = Double.parseDouble(s);
+        s = pos.substring(degChars);
+        double min = Double.parseDouble(s);
+        deg = deg + min / 60;
+        if (direction.contains("S") || direction.contains("W")) {
+            deg = -deg;
         }
-        return floatLatDegrees;
+        return deg;
     }
 
-    double convertPositionlon(String pos, String direction) {
-        double floatLonDegrees = 0;
-        double floatLonMinutes = 0;
-        String s;
-
-        if (pos.length() < 6) {
-            return 0;
-        }
+    double convertPositionlat(String pos, String direction) {
+        return convertPosition(pos, direction, 2);
+    }
 
-        // This copies the arrays to temporary arrays
-        s = pos.substring(0, 3);
-        floatLonDegrees = Double.parseDouble(s);
-        s = pos.substring(3);
-        floatLonMinutes = Double.parseDouble(s);
-        floatLonDegrees = floatLonDegrees + floatLonMinutes / 60;
-        if (direction.contains("W")) {
-            floatLonDegrees = floatLonDegrees * -1;
-        }
-        return floatLonDegrees;
+    double convertPositionlon(String pos, String direction) {
+        return convertPosition(pos, direction, 3);
     }
 
     public String get_timeNmea() {
```

Comment:
can you also fix these '_' in this and following methods? Also unless it exceeds the 1000 LoC limit, can you please remove the "m_" from the class fields